pkg/sqlmigrations/migrations.go

```diff
@@ -512,3 +516,38 @@ func addRootUser(ctx context.Context, r runner) error {
 	}
 	return err
 }
+
+func addRoles(ctx context.Context, r runner) error {
+	// System tables can only be modified by a privileged internal user.
+	session := r.newRootSession(ctx)
+	defer session.Finish(r.sqlExecutor)
+
+	// Add the roles column to the system.users table.
+	const alterStmt = `
+		ALTER TABLE system.users ADD COLUMN IF NOT EXISTS "isRole" BOOL DEFAULT false;
+		 `
+
+	if res, err := r.sqlExecutor.ExecuteStatementsBuffered(
+		session, alterStmt, nil /* pinfo */, 1 /* expectedNumResults */); err == nil {
+		res.Close(ctx)
+	} else {
+		return err
+	}
+
+	// Create the `admin` role. We overwrite any existing user named "admin".
```

Comment:
@vivekmenezes: how should this be handled? The `UPSERT` should really wait until the `isRole` column exists, but we also don't want really long-running migrations.
This seems to be a problem with migrations in general, we don't seem to have a good way of having schema changes and data insertion that need the new schema.

Reply:
+ @a-robinson who did the initial migration work.

Reply:
Hmm, how long does it take for the `isRole` column to be usable? There's nothing inherently wrong with one of these migrations taking a while, because the node doesn't accept incoming SQL connections until the migrations are done. I'd say users might be surprised by a wait during startup, but since they'd only see this during an upgrade, it could be documented and probably wouldn't be too shocking.

Reply:
It shouldn't be too long, but we still need to pick a timeout or max retries.
There's also the fact that right now retrying the `upsert` within the migration fails some completely unrelated tests pretty hard. I'm trying to figure out if it's a problem with the tests or with what I'm doing.